### DISCUSSION TITLE
Publish to npm with provenance

### DISCUSSION THREAD

**Discussion by gabibguti** [storybookjs/storybook#23916] [Ideas]
### Is your feature request related to a problem? Please describe.

This is a request for adopting a security practice of generating provenance.

npm has released a new feature to [generate provenance](https://docs.npmjs.com/generating-provenance-statements) when publishing packages. The provenance is a file that holds verifiable information about the software artifacts describing where, when and how it was produced. Publishing provenance helps both the users and the maintainers be more confident that the package was published from the original repository by the original authors and not a malicious actor.

### Describe the solution you'd like

Add the `--provenance` flag to your publish command at [scripts/release/publish.ts#L127](https://github.com/storybookjs/storybook/blob/998926f115259ffe4e9afe03b25daf34556e4756/scripts/release/publish.ts#L127).

### Describe alternatives you've considered

None.

### Are you able to assist to bring the feature to reality?

yes, I can

### Additional context

Hi! I'm Gabriela and I work on behalf of Google and the OpenSSF suggesting supply-chain security changes :)

**Comment by shilman** [comment]
This sounds great. Any chance you can put together a PR with the change?

**Reply by gabibguti** [reply]
Sure!

**Reply by gabibguti** [reply]
For reference: https://github.com/storybookjs/storybook/pull/23917
